File: app/src/main/java/vasuki/istanpdf/data/DocumentManager.java
package vasuki.istanpdf.data;

import android.content.Context;
import android.database.Cursor;
import android.net.Uri;
import android.provider.OpenableColumns;

public final class DocumentManager {
    private final Context context;

    public DocumentManager(Context context) {
        this.context = context.getApplicationContext();
    }

    public String getDisplayName(Uri uri) {
        if (uri == null) return "IStanPdf";
        try (Cursor cursor = context.getContentResolver().query(uri, null, null, null, null)) {
            if (cursor != null && cursor.moveToFirst()) {
                int index = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
                if (index != -1) {
                    String name = cursor.getString(index);
                    int dot = name.lastIndexOf('.');
                    if (dot > 0) return name.substring(0, dot);
                    return name;
                }
            }
        } catch (Exception ignored) {
        }
        return "IStanPdf";
    }
}

File: app/src/main/kotlin/vasuki/istanpdf/data/DocumentManager.kt
package vasuki.istanpdf.data

import android.content.Context
import android.net.Uri
import android.provider.OpenableColumns

class DocumentManager(context: Context) {
    private val context = context.applicationContext

    fun getDisplayName(uri: Uri?): String {
        if (uri == null) return "IStanPdf"
        return try {
            context.contentResolver.query(uri, null, null, null, null)?.use { cursor ->
                if (cursor.moveToFirst()) {
                    val index = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
                    if (index != -1) {
                        val name = cursor.getString(index)
                        val dot = name.lastIndexOf('.')
                        return if (dot > 0) name.substring(0, dot) else name
                    }
                }
                null
            } ?: "IStanPdf"
        } catch (e: Exception) {
            "IStanPdf"
        }
    }
}
